kedro/config/omegaconf_config.py

```diff
@@ -147,15 +161,21 @@ def __getitem__(self, key) -> Dict[str, Any]:
         read_environment_variables = key == "credentials"
 
         # Load base env config
-        base_path = str(Path(self.conf_source) / self.base_env)
+        if self._protocol == "file":
+            base_path = str(Path(self.conf_source) / self.base_env)
+        else:
+            base_path = str(Path(self._fs.ls("")[-1]) / self.base_env)
         base_config = self.load_and_merge_dir_config(
             base_path, patterns, read_environment_variables
         )
         config = base_config
 
         # Load chosen env config
         run_env = self.env or self.default_run_env
-        env_path = str(Path(self.conf_source) / run_env)
+        if self._protocol == "file":
+            env_path = str(Path(self.conf_source) / run_env)
+        else:
+            env_path = str(Path(self._fs.ls("")[-1]) / run_env)
```

Comment:
Does it works for everyone? I get a Type error here. The unit tests seem working fine so I am not sure...

I am using fsspec==2023.3.0

`TypeError: expected str, bytes or os.PathLike object, not dict`

![image](https://user-images.githubusercontent.com/18221871/223797370-bdfebc4e-86db-4d9a-8d4b-68836f9eb903.png)

Reply:
Did some digging, I think we need to add `details=False` here. This recent PR in `fsspec` change the defaults to `True`.

https://github.com/fsspec/filesystem_spec/pull/1162